tests/jballerina-unit-test/src/test/resources/test-src/annotations/external_dependency_annotation.bal

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+// Functions
+@java:ExternalDependency
+function foo(int x) returns int {
+    return x * 2;
+}
+
+function bar() returns string {
+    return "bar";
+}
+
+// Attached functions
+class CorgeClass {
+
+    @java:ExternalDependency
+    function externalCorge() returns string {
+        return "external corge";
+    }
+
+    function corge() returns string {
+        return "corge";
+    }
+}
+
+// Classes
+@java:ExternalDependency
+class BazClass {
+    int x = 0;
+    string y = "";
+}
+
+class QuzClass {
+    int x = 0;
+    string y = "";
+}
+
+// Records
+@java:ExternalDependency
+type FooRec record {
+    int x;
+    string y;
+};
+
+type BarRec record {
+    int x;
+    string y;
+};
+
```

Comment:
Let's add an example with `object` type as well.

Reply:
Added in f65fc671efb24f12f870bca93183131c00a9e718.